server/neptune/workflows/internal/prrevision/workflow.go

```diff
@@ -119,19 +125,49 @@ func (r *Runner) setRevision(ctx workflow.Context, req Request, prs []github.Pul
 	return nil
 }
 
+func (r *Runner) listModifiedFilesAsync(ctx workflow.Context, req Request, prs []github.PullRequest) map[github.PullRequest]workflow.Future {
+	now := workflow.Now(ctx)
```

Comment:
Why are you doing this here instead of when you're using it?  Im not sure this is valid if a workflow can take hours/cross day boundaries.

Reply:
> Why are you doing this here instead of when you're using it? 

Since `workflow.Now()` calls adds an event to the event history, if we make this call everytime we process an open PR, the event history would be cluttered with these calls imo that's why I chose to precompute this and pass it along. 

> Im not sure this is valid if a workflow can take hours/cross day boundaries.

We're using a timestamp and passing it along to compare with the timestamp of a PR right? Why would this be a problem? If the workflow execution needs to be rescheduled on another worker after we compute this time, it'll still be in the event history. The only issue I see is if the workflow execution gets stuck for days after computing the time, in which case our computation for `isPrUpdatedWithinDays` might be off by a few days, but I doubt we'd run into that scenario. 

Lmk if you think I'm missing something